src/caf/toolkit/io.py

```diff
@@ -229,3 +230,68 @@ def read_csv_matrix(
             matrix = matrix.reindex(matrix.columns, axis=0)
 
     return matrix
+
+
+def find_file(
+    folder: pathlib.Path, name: str, suffixes: collections.abc.Sequence[str]
+) -> pathlib.Path:
+    """Find file based on acceptable suffixes.
+
+    Parameters
+    ----------
+    folder
+        Folder to search for file with, doesn't search within sub-folders.
+    name
+        Filename to search for, this should **not** include suffixes
+        (file extensions).
+    suffixes
+        Allowed suffixes to find, if multiple files are found with
+        acceptable `suffixes` then the one with the suffix first
+        in `suffixes` is returned.
+
+    Returns
+    -------
+    pathlib.Path
+        First file found in list of `suffixes`.
+
+    Raises
+    ------
+    FileNotFoundError
+        If no file can be found with `suffixes`.
+
+    Warns
+    -----
+    RuntimeWarning
+        If multiple files are found with the same name but different suffixes.
+    """
+    found: list[pathlib.Path] = []
+    unexpected: list[str] = []
+
+    for path in folder.glob(f"{name}.*"):
+        suffix = "".join(path.suffixes)
```

Comment:
Think a comment is needed here to explain why the join is needed. Guessing below

```suggestion
        # Combines multiple suffixes into one, does nothing if only one suffix exists
        suffix = "".join(path.suffixes)
```